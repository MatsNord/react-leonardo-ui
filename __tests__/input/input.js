import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../../src/input';

test('Normal input renders correctly', () => {
  const tree = renderer.create(
    <Input value="Value" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Input with states renders correctly', () => {
  const tree = renderer.create(
    <Input value="Value" invalid disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Input with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Input value="Value" variant="inverse" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Input passes props correctly', () => {
  const onClick = jest.fn(() => {});

  const tree = renderer.create(
    <Input value="Value" className="my-class" data-x="x" title="Title" onClick={onClick} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  expect(onClick.mock.calls.length).toBe(1);
});
