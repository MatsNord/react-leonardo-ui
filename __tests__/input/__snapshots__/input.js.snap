exports[`test Input passes props correctly 1`] = `
<input
  className="lui-input my-class"
  data-x="x"
  onClick={[Function]}
  title="Title"
  value="Value" />
`;

exports[`test Input with modifiers renders correctly 1`] = `
<input
  className="lui-input lui-input--inverse"
  value="Value" />
`;

exports[`test Input with states renders correctly 1`] = `
<input
  className="lui-input lui-invalid lui-disabled"
  value="Value" />
`;

exports[`test Normal input renders correctly 1`] = `
<input
  className="lui-input"
  value="Value" />
`;
